Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java

```diff
@@ -323,4 +365,182 @@ private static ListCertificatesResponseMessageType buildListCertificatesResponse
     private static KeyStore getTrustStore() {
         return CertificateManagerImpl.getInstance().getTrustStore();
     }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see gov.hhs.fha.nhinc.configadmin.EntityConfigAdminPortType#createCSR(gov.hhs.fha.nhinc.common.configadmin.
+     * CreateCSRRequestMessageType)
+     */
+    @Override
+    public SimpleCertificateResponseMessageType createCSR(CreateCSRRequestMessageType request) {
+
+        String alias = request.getAlias();
+        if(StringUtils.isBlank(alias)){
+            return buildSimpleResponse(false, "alias is required.");
+        }
+
+        // keytool -certreq -alias gateway -keystore gateway.jks -file gateway-yyyyMMdd.csr
+        try {
+            ByteArrayDataSource dsCsr = new ByteArrayDataSource(
+                getPemCsrBy(getGatewayNew(), System.getProperty(KEY_STORE_PASSWORD_KEY), alias), MIME_TEXT_PLAIN);
+            SimpleCertificateResponseMessageType response = buildSimpleResponse(true, ACT_SUCCESSFUL);
+            response.setCsrData(new DataHandler(dsCsr));
+            return response;
+        } catch (Exception e) {
+            LOG.error("error while generating the csr: {}", alias, e);
+            return buildSimpleResponse(false, "fail to generate the csr: " + alias);
+        }
+
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * gov.hhs.fha.nhinc.configadmin.EntityConfigAdminPortType#createCertificate(gov.hhs.fha.nhinc.common.configadmin.
+     * CreateCertificateRequestMessageType)
+     */
+    @Override
+    public SimpleCertificateResponseMessageType createCertificate(CreateCertificateRequestMessageType request) {
+
+        String alias = request.getAlias();
+        String ou = request.getOrganizationalUnit();
+        String o = request.getOrganization();
+        String c = request.getCountryName();
+        String refNumber = request.getReferenceNumber();
+        if (StringUtils.isBlank(alias) || StringUtils.isBlank(ou) || StringUtils.isBlank(o) || StringUtils.isBlank(c)
+            || StringUtils.isBlank(refNumber)) {
+            return buildSimpleResponse(false,
+                "alias, Organizational Unit, Organization, Country Name and Reference Number are required.");
```

Comment:
"Alias" if we are capitalizing everything else here. Comma after "Country Name" as well.